pkg/scheduler/apis/config/v1alpha1/defaults_test.go

```diff
@@ -29,112 +29,160 @@ import (
 )
 
 var _ = Describe("Defaults", func() {
-	Describe("SchedulerConfiguration", func() {
-		var obj *schedulerv1alpha1.SchedulerConfiguration
+	var obj *schedulerv1alpha1.SchedulerConfiguration
 
-		BeforeEach(func() {
-			obj = &schedulerv1alpha1.SchedulerConfiguration{}
-		})
+	BeforeEach(func() {
+		obj = &schedulerv1alpha1.SchedulerConfiguration{}
+	})
 
-		Context("Empty configuration", func() {
-			It("should correctly default the admission controller configuration", func() {
-				schedulerv1alpha1.SetObjectDefaults_SchedulerConfiguration(obj)
+	Describe("SchedulerConfiguration defaulting", func() {
+		It("should default the admission controller configuration", func() {
```

Comment:
You could extract the Scheduler configuration defaulting to a new defaulting function
``` diff
diff --git a/pkg/scheduler/apis/config/v1alpha1/defaults.go b/pkg/scheduler/apis/config/v1alpha1/defaults.go
index 602546542..cdefaade1 100644
--- a/pkg/scheduler/apis/config/v1alpha1/defaults.go
+++ b/pkg/scheduler/apis/config/v1alpha1/defaults.go
@@ -28,28 +28,29 @@ func SetDefaults_SchedulerConfiguration(obj *SchedulerConfiguration) {
 		obj.LogFormat = LogFormatJSON
 	}
 
-	if obj.Schedulers.BackupBucket == nil {
-		obj.Schedulers.BackupBucket = &BackupBucketSchedulerConfiguration{
-			ConcurrentSyncs: 2,
-		}
+	if obj.LeaderElection == nil {
+		obj.LeaderElection = &componentbaseconfigv1alpha1.LeaderElectionConfiguration{}
 	}
+}
 
-	if obj.Schedulers.Shoot == nil {
-		obj.Schedulers.Shoot = &ShootSchedulerConfiguration{
-			ConcurrentSyncs: 5,
-			Strategy:        Default,
-		}
+func SetDefaults_SchedulerControllerConfiguration(obj *SchedulerControllerConfiguration) {
+	if obj.BackupBucket == nil {
+		obj.BackupBucket = &BackupBucketSchedulerConfiguration{}
 	}
-	if len(obj.Schedulers.Shoot.Strategy) == 0 {
-		obj.Schedulers.Shoot.Strategy = Default
+
+	if obj.BackupBucket.ConcurrentSyncs == 0 {
+		obj.BackupBucket.ConcurrentSyncs = 2
 	}
 
-	if obj.Schedulers.Shoot.ConcurrentSyncs == 0 {
-		obj.Schedulers.Shoot.ConcurrentSyncs = 5
+	if obj.Shoot == nil {
+		obj.Shoot = &ShootSchedulerConfiguration{}
+	}
+	if len(obj.Shoot.Strategy) == 0 {
+		obj.Shoot.Strategy = Default
 	}
 
-	if obj.LeaderElection == nil {
-		obj.LeaderElection = &componentbaseconfigv1alpha1.LeaderElectionConfiguration{}
+	if obj.Shoot.ConcurrentSyncs == 0 {
+		obj.Shoot.ConcurrentSyncs = 5
 	}
 }
 

```